src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -0,0 +1,217 @@
+import { html, LitElement } from "@polymer/lit-element";
+import "@polymer/paper-checkbox/paper-checkbox.js";
+import "@polymer/paper-icon-button/paper-icon-button.js";
+import "@polymer/paper-input/paper-input.js";
+import "@polymer/paper-item/paper-icon-item.js";
+import "@polymer/paper-item/paper-item-body.js";
+
+import "../../../components/ha-card.js";
+
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../types.js";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+interface Config extends LovelaceConfig {
+  title?: string;
+}
+
+interface ListItem {
+  id: number;
+  name: string;
+  complete: boolean;
+}
+
+class HuiShoppingListCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config?: Config;
+  private items?: ListItem[];
+
+  static get properties() {
+    return {
+      hass: {},
+      config: {},
+      items: [],
+    };
+  }
+
+  public getCardSize() {
+    return 1;
+  }
+
+  public setConfig(config: Config) {
+    this.config = config;
+    this.items = [];
+  }
+
+  protected render() {
+    if (!this.config || !this.hass) {
+      return html``;
+    }
+
+    // TODO Localize 'Add Item' text
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this.config.title}">
+        <div class="addRow">
+          <paper-input
+            id='addBox'
+            placeholder='Add Item'
```

Comment:
`.placeholder`

When you're rendering a web component, prefer properties over attributes.

Reply:
Except for when you need it for aria or CSS, things that require attributes

Reply:
Out of curiosity do you have any references on why we should prefer properties to attributes? Would like to read more on any advantages or the reasons behind it. Thanks.

Reply:
`.placeholder` doesn't work either